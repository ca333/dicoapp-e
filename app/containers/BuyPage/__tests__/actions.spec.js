import { loadPrice, loadPrices, loadPricesSuccess } from '../actions';
import { LOAD_PRICE, LOAD_PRICES, LOAD_PRICES_SUCCESS } from '../constants';

describe('BuyPage Actions', () => {
  describe('loadPrice action', () => {
    it('should loadPrice should create loadPrice action', () => {
      const coin = 'coin';
      expect(loadPrice(coin)).toMatchSnapshot();
    });

    it('should return the correct type and the passed name', () => {
      const coin = 'coin';
      const expectedResult = {
        type: LOAD_PRICE,
        payload: {
          coin
        }
      };

      expect(loadPrice(coin)).toEqual(expectedResult);
    });
  });

  describe('loadPrices action', () => {
    it('should loadPrices should create loadPrices action', () => {
      expect(loadPrices()).toMatchSnapshot();
    });

    it('should return the correct type and the passed name', () => {
      const expectedResult = {
        type: LOAD_PRICES
      };

      expect(loadPrices()).toEqual(expectedResult);
    });
  });

  describe('loadPricesSuccess action', () => {
    it('should loadPricesSuccess should create loadPricesSuccess action', () => {
      expect(loadPricesSuccess()).toMatchSnapshot();
    });

    it('should return the correct type and the passed name', () => {
      const expectedResult = {
        type: LOAD_PRICES_SUCCESS
      };

      expect(loadPricesSuccess()).toEqual(expectedResult);
    });
  });
});
