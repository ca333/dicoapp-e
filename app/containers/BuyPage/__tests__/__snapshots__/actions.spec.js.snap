// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuyPage Actions loadPrice action should loadPrice should create loadPrice action 1`] = `
Object {
  "payload": Object {
    "coin": "coin",
  },
  "type": "dicoapp/BuyPage/LOAD_PRICE",
}
`;

exports[`BuyPage Actions loadPrices action should loadPrices should create loadPrices action 1`] = `
Object {
  "type": "dicoapp/BuyPage/LOAD_PRICES",
}
`;

exports[`BuyPage Actions loadPricesSuccess action should loadPricesSuccess should create loadPricesSuccess action 1`] = `
Object {
  "type": "dicoapp/BuyPage/LOAD_PRICES_SUCCESS",
}
`;
